app/src/androidTest/java/com/github/sdpteam15/polyevents/UpcomingActivitiesHomeFragmentTest.kt

```diff
@@ -0,0 +1,75 @@
+package com.github.sdpteam15.polyevents
+
+
+import androidx.test.espresso.Espresso
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.hasChildCount
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import com.github.sdpteam15.polyevents.activity.Activity
+import com.github.sdpteam15.polyevents.helper.ActivitiesQueryHelperInterface
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito.`when`
+import org.mockito.Mockito.mock
+import org.mockito.junit.MockitoJUnitRunner
+import java.time.LocalDateTime
+
+@RunWith(MockitoJUnitRunner::class)
+class UpcomingActivitiesHomeFragmentTest {
+
+    lateinit var activities: ArrayList<Activity>
+    lateinit var mockedUpcomingActivitiesProvider: ActivitiesQueryHelperInterface
+
+    @Rule
+    @JvmField
+    var mainActivity = ActivityScenarioRule(MainActivity::class.java)
+
+    @Before
+    fun setup() {
+        activities = ArrayList<Activity>()
+
+        activities.add(
+            Activity(
+                "Sushi demo",
+                "Super hungry activity !",
+                LocalDateTime.of(2021, 3, 7, 12, 15),
+                1F,
+                "The fish band",
+                "Kitchen", null, "1")
+        )
+
+        activities.add(
+            Activity(
+                "Aqua Poney",
+                "Super cool activity !",
+                LocalDateTime.of(2021, 3, 7, 15, 0),
+                1.5F,
+                "The Aqua Poney team",
+                "Swimming pool", null, "2")
+        )
+
+        activities.add(
+            Activity(
+                "Concert",
+                "Super noisy activity !",
+                LocalDateTime.of(2021, 3, 7, 21, 15),
+                2.75F,
+                "AcademiC DeCibel",
+                "Concert Hall", null, "3"))
+
+        mockedUpcomingActivitiesProvider = mock(ActivitiesQueryHelperInterface::class.java)
+        `when`(mockedUpcomingActivitiesProvider.getUpcomingActivities()).thenReturn(activities)
```

Comment:
mockedUpcomingActivitiesProvider, never use in the test,
if you will change the the number of  activities in "activities" it will fail because ti will take the data in ActivitiesQueryHelper